tests/test_event_loops.py

```diff
@@ -2,31 +2,63 @@
 
 import asyncio
 import os
+import socket
 import sys
 import typing
 import unittest
 
 import urwid
 
+if typing.TYPE_CHECKING:
+    from types import TracebackType
+
+IS_WINDOWS = os.name == "nt"
+
+
+class ClosingSocketPair(typing.ContextManager[typing.Tuple[socket.socket, socket.socket]]):
```

Comment:
https://docs.python.org/3/library/select.html?highlight=select#select.select
```
File objects on Windows are not acceptable, but sockets are. On Windows, the underlying select() function is provided by the WinSock library, and does not handle file descriptors that don’t originate from WinSock.
```
Ideally we need to get rid of `select` usage